### PR TITLE
Build pkgsrc packages

### LINKED ISSUE #321: SmartOS can't install SVR4-style Solaris packages
Let me try this again in the right repo...

SmartOS does not have the old Solaris `pkgadd` commands as it uses BSD pkgsrc under the covers. Thus, our attempt to coerce it into installing a SVR4-type .pkg file is actually incorrect.

If we actually do intend to support SmartOS in the context of the platform support RFC, then we would need to build native pkgsrc packages for it. (You could `pkgtrans` the Solaris ones to it as part of CI or something, as a cheap trick.)

### PR DESCRIPTION
**Pull request by thommay**
For now, only tested on SmartOS. Fixes #321 

--------------------------------------------------
/cc @chef/omnibus-maintainers @juliandunn

### CODE REVIEW

**Comment by thommay** [comment]
(I can't restart appveyor but it's SF timeout related)

File: lib/omnibus/health_check.rb [hunk 1]
```diff
@@ -124,6 +124,12 @@ class HealthCheck
       /libgcc_s.so/,
       /libstdc\+\+\.so/,
       /libcrypt.so/,
+      /libcrypto.so/,
+      /libssl.so/,
```
Comment (stevendanna) [inline lib/omnibus/health_check.rb]: These are a bit worrying.  Shouldn't these always be safely provided by OpenSSL?

Comment (scotthain) [inline lib/omnibus/health_check.rb]: These are an unfortunate side-effect of Solaris 11's Cryptographic Framework, which kind of includes ssl in the kernel. If you're morbidly curious: https://docs.oracle.com/cd/E23824_01/html/821-1456/scf-10.html#scrolltoc

**Comment by thommay** [comment]
Actually for reasons I don't understand, there seems to be something wonky with SSL:
```
[root@motet ~]# knife ssl check -c /etc/chef/client.rb
Connecting to host api.chef.io:443
ERROR: The SSL certificate of api.chef.io could not be verified
Certificate issuer data: /C=US/O=DigiCert Inc/CN=DigiCert SHA2 Secure Server CA
```

like it's not using the omnibus trust store?

**Comment by scotthain** [comment]
@thommay I've never seen that one before... I might suggest commenting out all the chef defs in your project file and add a curl dep, so you can debug the SSL through curl. I've had some success with that in the past.

**Comment by thommay** [comment]
The ssl thing appears transient, so i'm gonna merge

**Pull request merged**